### PR TITLE
Fix issue #60 - add new tests for core app

### PR DESCRIPTION
**Pull request by maxsocl**
@cjlee112 @dixon-che @VladimirFilonov 

Add new tests for ``ct`` app and refactor old tests.

Issue https://github.com/cjlee112/socraticqs2/issues/60

### CODE REVIEW

**Comment by sengottuvelk** [comment]
@maxsocl 
There are places where the docstrings given as `'docstring'`. We might want change them as `'''docstring'''` or `""""docstring"""`. Is there any reason behind it for the way it is now?

File: mysite/ct/tests/integrate.py [hunk 1]
```diff
@@ -0,0 +1,492 @@
...
+        self.assertEqual(ul2b.lesson.text, 'Big bad wolf')
+        self.assertFalse(self.ul.lesson.pk == ul2b.lesson.pk)
+        self.assertEqual(Concept.objects.get(conceptlink__lesson__unitlesson=ul2b),
+                         self.ul.lesson.concept)
+        self.assertTrue(ul2b.lesson.is_committed())
+
+
+class FakeRequest(object):
+    'trivial holder for request data to pass to test calls'
+    def __init__(self, user, sessionDict=None, method='POST', dataDict=None,
```
Comment (sengottuvelk) [inline mysite/ct/tests/integrate.py]: Instead checking for `sessionDict ` for `None`, set defualt value to `sessionDict={}`

File: mysite/test/factories/user.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import factory
+from django.contrib.auth.models import User
+
+
+class UserFactory(factory.django.DjangoModelFactory):
+    class Meta:
+        model = User
+        django_get_or_create = ('username', 'email')
+
+    username = 'john.doe'
```
Comment (sengottuvelk) [inline mysite/test/factories/user.py]: Instead hard-coding the values for the factory, we can make use of `fake-factory` to generate random values each times. Refer the [documentation for `fake-factory`](http://fake-factory.readthedocs.org/en/latest/)
The code might look like,

```python
from faker import Factory as FakerFactory
faker = FakerFactory.create()

class UserFactory(factory.django.DjangoModelFactory):
    class Meta:
        model = User

    username = factory.LazyAttribute(lambda x: faker.user_name())
    email = factory.LazyAttribute(lambda x: faker.email())
```

**Comment by sengottuvelk** [comment]
Otherwise LGTM, Merge when ready.

**Comment by maxsocl** [comment]
> @maxsocl 
> There are places where the docstrings given as 'docstring'. We might want change them as '''docstring''' or """docstring""". or Is there any reason behind it for the way it is now?

Refactored.

File: mysite/test/factories/user.py (hunk 2 above)
Comment (maxsocl) [inline mysite/test/factories/user.py]: > Instead hard-coding the values for the factory, we can make use of fake-factory to generate random values each times. Refer the documentation for fake-factory
> The code might look like,
> 
> from faker import Factory as FakerFactory
> faker = FakerFactory.create()
> 
> class UserFactory(factory.django.DjangoModelFactory):
>     class Meta:
>         model = User
> 
>     username = factory.LazyAttribute(lambda x: faker.user_name())
>     email = factory.LazyAttribute(lambda x: faker.email())

Yes, then is great if I need to create 50 users w/ random names. At this point I need only one user and it is ok to use the same name. Also I do not want to add another one external dependancy now. May be for future complex test we can add this proposal.
BTW thanks for good comment!

**Pull request merged**